types/request.d.ts

```diff
@@ -38,7 +38,35 @@ export interface RequestRouteOptions<ContextConfig = ContextConfigDefault, Schem
  * FastifyRequest is an instance of the standard http or http2 request objects.
  * It defaults to http.IncomingMessage, and it also extends the relative request object.
  */
-export interface FastifyRequest<RouteGeneric extends RouteGenericInterface = RouteGenericInterface,
+export type FastifyRequest<
+  RouteGeneric extends RouteGenericInterface = RouteGenericInterface,
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
+  SchemaCompiler extends FastifySchema = FastifySchema,
+  TypeProvider extends FastifyTypeProvider = FastifyTypeProviderDefault,
+  ContextConfig = ContextConfigDefault,
+  Logger extends FastifyBaseLogger = FastifyBaseLogger,
+  Decorators extends FastifyDecorators['request'] = FastifyDecorators['request'],
+  RequestType extends FastifyRequestType = ResolveFastifyRequestType<TypeProvider, SchemaCompiler, RouteGeneric>
```

Comment:
You inserted the new generic between existing ones `Logger` and `RequestType`, why?
https://github.com/fastify/fastify/blob/2ee97ffa57f80be390414585ca8c07ecdaf250bc/types/request.d.ts#L47

Reply:
Because RequestType had a comment which lead me to believe that it is supposed to be last no matter the number of other params.

> // ^ Temporary Note: RequestType has been re-ordered to be the last argument in
//   generic list. This generic argument is now considered optional as it can be
//   automatically inferred from the SchemaCompiler, RouteGeneric and TypeProvider
//   arguments. Implementations that already pass this argument can either omit
//   the RequestType (preferred) or swap Logger and RequestType arguments when
//   creating custom types of FastifyRequest. Related issue #4123

Since it is supposed to be automatically inferred by design i left it last, otherwise it would be necessary to explicitly declare it which would've been a breaking change the way i understood the design.

 That said, i don't have an issue with moving Decorators down, but i'm not sure that's the best thing to do.

TypeScript doesn't do partial inference, so i'm not sure if it's a good idea to parameterize the request "manually" to begin with, because at that point it's all or nothing.

Reply:
My bad, I didn't catch it.